minitwit/src/main/java/WebApplication.java

```diff
@@ -665,16 +668,69 @@ else if (!BCrypt.checkpw(enteredPW, rs.getString("pw_hash"))) {
         updateLatest(request);
 
         var messages = getMessages();
-        var filteredMessages = messages.stream().map((message) -> {
-            return Map.ofEntries(
-                    Map.entry("content", message.get("text")),
-                    Map.entry("pub_date", message.get("pub_date")),
-                    Map.entry("user", message.get("username"))
-            );
-        });
+        var filteredMessages = messages.stream().map((message) -> Map.ofEntries(
+               Map.entry("content", message.get("text")),
+               Map.entry("pub_date", message.get("pub_date")),
+               Map.entry("user", message.get("username"))
+       ));
         return filteredMessages.toList();
     };
 
+    //  --------------------------------------
+    //              Simulator API
+    //  ---------------------------------------
+
+    public static Route serveSimMsgsUsr = (Request request, Response response) -> {
+
+        updateLatest(request);
+
+        SQLite db = new SQLite();
+        var connection = db.getConnection();
+        var userID = getUserID(db, request.params(":username"));
+
+        int noMsgs =  Integer.parseInt(request.queryParamOrDefault("no", "100"));
+
+        if (request.requestMethod().equals("GET")){
+            if (userID == 0) {
+                response.status(404);
```

Comment:
connection is not closed here.
Also, if we POST with a invalid user this one doesn't catch that, but that bug also exists in the spec api :D